src/main/kotlin/com/zepben/evolve/services/network/EquivalentNetworkUtils.kt

```diff
@@ -0,0 +1,443 @@
+        defaultEquivalentBranchCreator(branchMrid, initBranch),
+        defaultEquivalentEquipmentCreator(equipmentMrid, initEquipment),
+        maxNumber
+    )
+
+    /**
+     * A default EquivalentBranch creator that should cover most use cases. If you need to override this you should call
+     * addToEdgeBetweenContainers directly.
+     */
+    inline fun defaultEquivalentBranchCreator(
+        crossinline branchMrid: (EquivalentBranchDetails) -> String = { (edgeEquipment) -> "${edgeEquipment.mRID}-eb" },
+        crossinline initBranch: EquivalentBranch.(EquivalentBranchDetails) -> Unit = {},
+    ): (EquivalentBranchDetails) -> Sequence<Pair<PhaseCode, EquivalentBranch>> =
+        { (edgeEquipment, edgeNode) ->
+            sequenceOf(
+                getPhaseCode(edgeNode) to EquivalentBranch(branchMrid(edgeEquipment via edgeNode))
+                    .apply {
+                        location = edgeEquipment.location
+                        baseVoltage = getEquivalentBaseVoltage(edgeEquipment, edgeNode)
+                        initBranch(edgeEquipment via edgeNode)
+                    }
+            )
+        }
+
+    /**
+     * A default EquivalentEquipment creator that should cover most use cases. If you need to override this you should call
+     * addToEdgeBetweenContainers directly.
+     */
+    inline fun <reified T : ConductingEquipment> defaultEquivalentEquipmentCreator(
+        crossinline equipmentMrid: (EquivalentEquipmentDetails) -> String = { (edgeEquipment) -> "${edgeEquipment.mRID}-eeq" },
+        crossinline initEquipment: T.(EquivalentEquipmentDetails) -> Unit = {},
+    ): (EquivalentEquipmentDetails) -> Sequence<Pair<PhaseCode, ConductingEquipment>> =
+        { (edgeEquipment, edgeNode, equivalentBranch) ->
+            sequenceOf(
+                getPhaseCode(edgeNode) to requireNotNull(
+                    T::class.primaryConstructor
+                        ?.call(equipmentMrid(edgeEquipment via edgeNode via equivalentBranch))
+                        ?.apply {
+                            location = edgeEquipment.location
+                            baseVoltage = getEquivalentBaseVoltage(edgeEquipment, edgeNode)
+                            initEquipment(edgeEquipment via edgeNode via equivalentBranch)
+                        }
+                )
+            )
+        }
+
+    /**
+     * Function to add 'equivalent networks' which are groups of [ConductingEquipment] connected through [EquivalentBranch]es that represent simplified networks.
+     * These 'equivalent networks' are attached to edge [ConnectivityNode]s, which are [ConnectivityNode]s right at the edge between [EquipmentContainer]s.
+     * These edge [ConnectivityNode] are identified by having a single [Terminal] that has a [ConductingEquipment] that belongs to both neighbouring [EquipmentContainer]s.
+     *
+     * This overload of the function takes two [EquipmentContainer] instances to identify the edge [ConnectivityNode]s.
```

Comment:
You probably want to add "between them" at the end here? Also, might be worth explicitly noting what happens if there are no edge nodes between the two containers in this scenario. I assume you just get back an empty result, but it'd be good to tell the user that.

Reply:
Yep, that's correct, added more info in the doco